_posts/2020-02-07-metal3-dev-env-install-deep-dive.md

```diff
@@ -0,0 +1,921 @@
+
+<br>
+Below it is verified that the cloud image files were downloaded successfully in the defined folder:
+
+```sh
+[alosadag@eko1 metal3-dev-env]$ ll /opt/metal3-dev-env/ironic/html/images
+total 920324
+-rw-rw-r--. 1 alosadag alosadag 942407680 Feb  3 12:39 CentOS-7-x86_64-GenericCloud-1907.qcow2
+-rw-rw-r--. 1 alosadag alosadag        33 Feb  3 12:39 CentOS-7-x86_64-GenericCloud-1907.qcow2.md5sum
+```
+
+Once the `images.sh` shared script is sourced, the following container images are pre-cached locally to the host in order to speed up things later. Below it is shown the code snippet in charge of that task:
+
+```sh
++ for IMAGE_VAR in IRONIC_IMAGE IPA_DOWNLOADER_IMAGE VBMC_IMAGE SUSHY_TOOLS_IMAGE DOCKER_REGISTRY_IMAGE
++ IMAGE=quay.io/metal3-io/ironic
++ sudo podman pull quay.io/metal3-io/ironic
+...
+....
+```
+The container image location of each one is defined by their respective variables: 
+<br>
+<br>
+
+| **Name of the variable** | **Default value**      |
+| VBMC_IMAGE | quay.io/metal3-io/vbmc |
+| SUSHY_TOOLS_IMAGE | quay.io/metal3-io/sushy-tools |
+| IPA_DOWNLOADER_IMAGE | quay.io/metal3-io/ironic-ipa-downloader	|
+| IRONIC_IMAGE | quay.io/metal3-io/ironic |
+| DOCKER_REGISTRY_IMAGE | docker.io/registry:latest |
+
+<br>
+
+> **Information:** In case it is expected to run self-modified container images, just overwrite the previous variables to match the right location. It is worth mentioning that a the container registry running as a privileged container in the host, so you can just upload there.
+
+<br>
+At this point, an Ansible role is run locally in order to complete the local configuration.
+
+```sh 
+ansible-playbook \
+  -e "working_dir=$WORKING_DIR" \
+  -e "virthost=$HOSTNAME" \
+  -i vm-setup/inventory.ini \
+  -b -vvv vm-setup/install-package-playbook.yml
+  ```
+
+This playbook imports two roles. One called `packages_installation`, which is in charge of installing a few more packages. The list of packages installed are listed as default Ansible variables [here](https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/packages_installation/defaults/main.yml). The other role is based on a the [fubarhouse.golang](https://galaxy.ansible.com/fubarhouse/golang) Ansible Galaxy role. It is in charge of installing and configuring the exact `golang` version `1.12.12` defined in an Ansible variable [here](https://github.com/metal3-io/metal3-dev-env/blob/9fa752b90ed58fdadcd52c246d3023766dfcb2dc/vm-setup/install-package-playbook.yml#L12)
+
+Once the playbook is finished, a pod called `ironic-pod` is created. Inside that pod, a *privileged* `ironic-ipa-downloader` container is started and attached to the host network. This container is in charge of downloading the [Ironic Python Agent](https://docs.openstack.org/ironic-python-agent/latest/) (IPA) files to a shared volume defined by `IRONIC_IMAGE_DIR`. This folder is exposed by the `ironic` container through HTTP. 
```

Comment:
good catch. Thank you. I rewritten the centos_install_requirements section like:

"Next step deals with cleaning ironic containers and **pods** that could be running in the host from failed deployments. This will ensure that there will be no issues when creating `ironic-pod` and `infra-pod` a little bit later in this first step.
"